### PR TITLE
Feature/rgb_cartesian

### PR DESCRIPTION
**Pull request by tegkhanna**
added eucledian rgb cartesian comparision to find nearest match.
much more accurate

### CODE REVIEW

File: colors.py [hunk 1]
```diff
@@ -49,3 +49,37 @@ def nearest_match(self):
                     min = abs(item[1] - color_int)
                     color = item[0]
             return "Nearest Match: " + color
+    def RGB_cartesian_match(self):
```
Comment (apoorvaeternity) [inline colors.py]: Please replace the else part of the nearest_match() function with this block.

File: colors.py [hunk 2]
```diff
@@ -49,3 +49,37 @@ def nearest_match(self):
...
+        #eucledian
+        color = None
+        min = 16777215  # FFFFFF
```
Comment (apoorvaeternity) [inline colors.py]: Let's replace min with some other variable name because min is already a built-in function.

File: colors.py [hunk 3]
```diff
@@ -49,3 +49,37 @@ def nearest_match(self):
...
+        for item in colors[self.colset].values():
+            colorCompareHex = "0x{:06x}".format(item[1])
```
Comment (apoorvaeternity) [inline colors.py]: Variable names should be lowercase, with words separated by underscores as necessary to improve readability.

What are we doing here?
I think we are converting the integer value to hex. If that's the case then we can drop this integer part of the item because we can obtain it directly from the key.
Will make that change in a different commit.
This is good for now.

File: colors.py [hunk 4]
```diff
@@ -49,3 +49,37 @@ def nearest_match(self):
...
+            if self.eucleanCalculate(self.value, colorCompareHex) < min:
+                min = self.eucleanCalculate(self.value, colorCompareHex)
+                color = item[0]
+        return "Nearest Match: " + color
+	
+    def eucleanCalculate(self, C1, C2):
```
Comment (apoorvaeternity) [inline colors.py]: Function and variable names should be lowercase, with words separated by underscores as necessary to improve readability as described in PEP8.

File: colors.py [hunk 5]
```diff
@@ -49,3 +49,37 @@ def nearest_match(self):
...
+        C2_B_int = int(C2[4:6], base=16)
+        #delta square
+        deltaR = (C1_R_int - C2_R_int)**2 
+        deltaG = (C1_G_int - C2_G_int)**2 
+        deltaB = (C1_B_int - C2_B_int)**2
+        r = (C1_R_int + C2_R_int)/2
+        deltaC = ((2 + r/256)*deltaR + 4*deltaG + (2 + (255 - r)/2)*deltaB)
+        return deltaC**0.5
+
+
```
Comment (apoorvaeternity) [inline colors.py]: Remove these extra lines at the end of file.

**Review by apoorvaeternity — CHANGES_REQUESTED** [review]
Look's good to me. Make the requested changes and it will be ready to get merged.

**Comment by tegkhanna** [comment]
changes applied. Up for review

**Comment by apoorvaeternity** [comment]
This looks great!!! Thanks for your contribution.
